### ISSUE TITLE
[Bug]: Assert error

### ISSUE THREAD

**Issue by mbledkowski** [puppeteer/puppeteer#8582]
### Bug description

Steps to reproduce the problem:

1. Run my script ([github.com/pluget/bukkitScraper](https://github.com/pluget/bukkitScraper/tree/main/resourcesData))
2. After a long while script crashes with the following error
  ```sh
Error
    at assert (/home/mble/Code/repos/pluget/bukkitScraper/resourcesData/node_modules/puppeteer/src/common/assert.ts:27:11)
    at FrameManager._FrameManager_onFrameAttached (/home/mble/Code/repos/pluget/bukkitScraper/resourcesData/node_modules/puppeteer/src/common/FrameManager.ts:377:11)
    at /home/mble/Code/repos/pluget/bukkitScraper/resourcesData/node_modules/puppeteer/src/common/FrameManager.ts:97:28
    at /home/mble/Code/repos/pluget/bukkitScraper/resourcesData/node_modules/puppeteer/lib/cjs/vendor/mitt/src/index.js:51:68
    at Array.map (<anonymous>)
    at Object.emit (/home/mble/Code/repos/pluget/bukkitScraper/resourcesData/node_modules/puppeteer/lib/cjs/vendor/mitt/src/index.js:51:43)
    at CDPSession.emit (/home/mble/Code/repos/pluget/bukkitScraper/resourcesData/node_modules/puppeteer/src/common/EventEmitter.ts:102:18)
    at CDPSession._onMessage (/home/mble/Code/repos/pluget/bukkitScraper/resourcesData/node_modules/puppeteer/src/common/Connection.ts:346:12)
    at Connection._Connection_onMessage (/home/mble/Code/repos/pluget/bukkitScraper/resourcesData/node_modules/puppeteer/src/common/Connection.ts:165:17)
    at WebSocket.<anonymous> (/home/mble/Code/repos/pluget/bukkitScraper/resourcesData/node_modules/puppeteer/src/node/NodeWebSocketTransport.ts:61:24)
  ```

I believe this is bug on puppeteer side not on mine. It crashes after making couple of thousands requests. My machine have 16 GB of ram, i5 skylake processor, and runs Arch Linux

### Puppeteer version

"puppeteer": "^15.0.0",     "puppeteer-extra": "^3.3.0",

### Node.js version

v18.3.0

### npm version

8.5.5

### What operating system are you seeing the problem on?

Linux

### Relevant log output

```shell
Error
    at assert (/home/mble/Code/repos/pluget/bukkitScraper/resourcesData/node_modules/puppeteer/src/common/assert.ts:27:11)
    at FrameManager._FrameManager_onFrameAttached (/home/mble/Code/repos/pluget/bukkitScraper/resourcesData/node_modules/puppeteer/src/common/FrameManager.ts:377:11)
    at /home/mble/Code/repos/pluget/bukkitScraper/resourcesData/node_modules/puppeteer/src/common/FrameManager.ts:97:28
    at /home/mble/Code/repos/pluget/bukkitScraper/resourcesData/node_modules/puppeteer/lib/cjs/vendor/mitt/src/index.js:51:68
    at Array.map (<anonymous>)
    at Object.emit (/home/mble/Code/repos/pluget/bukkitScraper/resourcesData/node_modules/puppeteer/lib/cjs/vendor/mitt/src/index.js:51:43)
    at CDPSession.emit (/home/mble/Code/repos/pluget/bukkitScraper/resourcesData/node_modules/puppeteer/src/common/EventEmitter.ts:102:18)
    at CDPSession._onMessage (/home/mble/Code/repos/pluget/bukkitScraper/resourcesData/node_modules/puppeteer/src/common/Connection.ts:346:12)
    at Connection._Connection_onMessage (/home/mble/Code/repos/pluget/bukkitScraper/resourcesData/node_modules/puppeteer/src/common/Connection.ts:165:17)
    at WebSocket.<anonymous> (/home/mble/Code/repos/pluget/bukkitScraper/resourcesData/node_modules/puppeteer/src/node/NodeWebSocketTransport.ts:61:24)
```

**Comment by jrandolf-google** [comment]
> I believe this is bug on puppeteer side not on mine. It crashes after making couple of thousands requests.

Please investigate the request that causes the crash and produce a script that minimally showcases the crash.

**Comment by kbourro** [comment]
I got same problem after upgrading to version 15

**Comment by kbourro** [comment]
The only fix i have found is to use --disable-site-isolation-trials argument

**Comment by CataAlamos** [comment]
I have the same problem despite having a version < 15

**Comment by OrKoN** [comment]
I believe this error should be fixed by the next release v16.0.0 (although I was not able to reproduce the issue using the repro, we had a similar error in other scenarios which I believe has the same root cause.

**Comment by OrKoN** [comment]
We just released v16.0.0. Would you give that version a try? As the version is expected to fix a very similarly looking bug, I am closing this one. Please feel free to re-open with a repro on v16.0.0 and later.
